components/director/internal/open_resource_discovery/client.go

```diff
@@ -69,25 +84,48 @@ func (c *client) FetchOpenResourceDiscoveryDocuments(ctx context.Context, app *m
 	}
 
 	docs := make([]*Document, 0)
+	mutex := sync.RWMutex{}
+	wg := sync.WaitGroup{}
+	queue := make(chan DocumentDetails)
+	var fetchDocError error = nil
```

Comment:
Since this `fetchDocError` is shared between parallel go routines, better make it a slice which aggregates each of the go routines' potential errors. Otherwise only the last error from some of the go routines will be persisted.

Also don't forget to guard the writing in the slice with mutex.